imports/plugins/included/search-mongo/server/methods/searchcollections.js

```diff
@@ -261,13 +261,48 @@ export function buildOrderSearch(cb) {
   }
 }
 
+export function buildAccountSearchRecord(accountId, updatedFields) {
+  Logger.debug("building account search record");
+  check(accountId, String);
+  check(updatedFields, Array);
+
+  const account = Accounts.findOne(accountId);
+  // let's ignore anonymous accounts
+  if (account && account.emails && account.emails.length) {
+    const accountSearch = {};
+
+    // Not all required fields are used in search
+    // We need to filter through fields that are used,
+    // and only update the search index if one of those fields were updated
+    // forceIndex is included to forceIndexing on startup, or when manually added
+    const searchableFields = ["forceIndex", "emails", "firstName", "lastName", "phone"];
+
+    const shouldRunIndex = updatedFields && updatedFields.some((r) => searchableFields.includes(r));
+
+    // If updatedField contains one of the searchableFields, run the indexing
+    if (shouldRunIndex) {
+      for (const field of requiredFields.accounts) {
+        if (transformations.accounts[field]) {
+          accountSearch[field] = transformations.accounts[field](account[field]);
+        } else {
+          accountSearch[field] = account[field];
+        }
+      }
+      AccountSearch.insert(accountSearch);
+      const rawAccountSearchCollection = AccountSearch.rawCollection();
+      rawAccountSearchCollection.createIndex({ shopId: 1, emails: 1 }).catch(handleIndexUpdateFailures);
```

Comment:
I don't see why we're creating this index every time we insert a record. Only needs to happen once on server startup.

Reply:
I think you're right, the code on line 309 should take care of this, it runs via a build job.